### PR TITLE
port parameter n

### LINKED ISSUE #4: Port parameter -n

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #4

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -36,11 +41,35 @@ func main() {
...
-			j -= 1
+		for j > 0 {
+			if !*numericParameter {
+				if bytes.Compare(dataSplit[j], dataSplit[j-1]) < 0 {
+					dataSplit[j], dataSplit[j-1] = dataSplit[j-1], dataSplit[j]
+					j -= 1
+				} else {
+					break
+				}
+			} else if *numericParameter {
```
Comment (Al2Klimov) [inline main.go]: What is this condition for?

File: main.go [hunk 2]
```diff
@@ -36,11 +41,35 @@ func main() {
...
+				if comparer1 < comparer2 {
+					dataSplit[j], dataSplit[j-1] = dataSplit[j-1], dataSplit[j]
+					j -= 1
+				} else {
+					break
+				}
+			}
 		}
 	}
+
```
Comment (Al2Klimov) [inline main.go]: That's not part of the issue.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/port-parameter-n-4`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol for all 4 cases:

* `< in` / `in`
* `-n` / no `-n`

**Comment by justinmueller05082000** [comment]
**Test protocol for** _port parameter n_**:**

**GO Code:**
```
package main

import (
	"bytes"
	"flag"
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

func main() {
	numericParameter := flag.Bool("n", false, "Sort numbers from lowest to highest")
	flag.Parse()

	var data []byte

	if len(flag.Args()) < 1 {
		content, err := ioutil.ReadAll(os.Stdin)
		if err != nil {
			fmt.Println(err)
			return
		}
		data = content
	} else {
		fileContent, readFileErr := ioutil.ReadFile(flag.Args()[0])
		if readFileErr != nil {
			fmt.Println(readFileErr)
			return
		}
		data = fileContent
	}

	if string(data) == "" {
		return
	}

	dataSplit := bytes.Split(data, []byte("\n"))
	lengthOfData := len(dataSplit)

	for i := 0; i < lengthOfData; i++ {
		j := i
		for j > 0 {
			if !*numericParameter {
				if bytes.Compare(dataSplit[j], dataSplit[j-1]) < 0 {
					dataSplit[j], dataSplit[j-1] = dataSplit[j-1], dataSplit[j]
					j -= 1
				} else {
					break
				}
			} else {
				comparer1, convertErr1 := strconv.Atoi(string(dataSplit[j]))
				if convertErr1 != nil {
					fmt.Println(convertErr1)
					return
				}
				comparer2, convertErr2 := strconv.Atoi(string(dataSplit[j-1]))
				if convertErr2 != nil {
					fmt.Println(convertErr2)
					return
				}
				if comparer1 < comparer2 {
					dataSplit[j], dataSplit[j-1] = dataSplit[j-1], dataSplit[j]
					j -= 1
				} else {
					break
				}
			}
		}
	}
	dataJoin := bytes.Join(dataSplit, []byte("\n"))
	fmt.Printf("%s\n", strings.TrimPrefix(string(dataJoin), "\n"))
}
```

**toSort.txt:**
```
11
23
54
4
8
436
7
2
```

The GO code is compared with the command line command sort.

**_______________________________**

**Case (1):**

**GO CODE:**
Terminal: `go run . -n toSort.txt `
Output:
```
2
4
7
8
11
23
54
436
```

**SORT COMMAND:**
Terminal: `sort -n toSort.txt `
Output:
```
2
4
7
8
11
23
54
436
```

**Conclusion(1):**
Both results are equal.

**_______________________________**

**Case(2):**

**GO CODE:**
Terminal: `go run . toSort.txt` 
Output:
```
11
2
23
4
436
54
7
8
```

**SORT COMMAND:**
Terminal: `sort toSort.txt `
Output:
```
11
2
23
4
436
54
7
8
```

**Conclusion(2):**
Both results are equal.

**_______________________________**

**Case(3):**

**GO CODE:**
Terminal: `go run . -n < toSort.txt `
Output:
```
2
4
7
8
11
23
54
436
```

**SORT COMMAND:**
Terminal: `sort -n < toSort.txt `
Output:
```
2
4
7
8
11
23
54
436
```

**Conclusion(3):**
Both results are equal.

**_______________________________**

**Case(4):**

**GO CODE:**
Terminal: `go run . < toSort.txt` 
Output:
```
11
2
23
4
436
54
7
8
```

**SORT COMMAND:**
Terminal: `sort < toSort.txt` 
Output:
```
11
2
23
4
436
54
7
8
```

**Conclusion(4):**
Both results are equal.

**_______________________________**

**Final Conclusion:**
The program does exactly the same as the command line command sort.

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**